data/src/main/java/org/apache/iceberg/data/RegistryBasedFileWriterFactory.java

```diff
@@ -0,0 +1,191 @@
+    this.positionDeleteRowSchema = positionDeleteRowSchema;
+    this.writeProperties = writeProperties != null ? writeProperties : ImmutableMap.of();
+    this.rowSchemaType = rowSchemaType;
+    this.equalityDeleteSchemaType = equalityDeleteSchemaType;
+    this.positionalDeleteSchemaType = positionalDeleteSchemaType;
+  }
+
+  protected S rowSchemaType() {
+    return rowSchemaType;
+  }
+
+  protected S equalityDeleteRowSchemaType() {
+    return equalityDeleteSchemaType;
+  }
+
+  protected S positionDeleteRowSchemaType() {
+    return positionalDeleteSchemaType;
+  }
+
+  @Override
+  public DataWriter<T> newDataWriter(
+      EncryptedOutputFile file, PartitionSpec spec, StructLike partition) {
+    EncryptionKeyMetadata keyMetadata = file.keyMetadata();
+    Map<String, String> properties = table.properties();
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+
+    try {
+      return DataFileServiceRegistry.writeBuilder(dataFileFormat, inputType, file)
+          .schema(dataSchema)
+          .set(properties)
+          .set(writeProperties)
+          .metricsConfig(metricsConfig)
+          .engineSchema(rowSchemaType())
+          .withSpec(spec)
+          .withPartition(partition)
+          .withKeyMetadata(keyMetadata)
+          .withSortOrder(dataSortOrder)
+          .overwrite()
+          .dataWriter();
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  @Override
+  public EqualityDeleteWriter<T> newEqualityDeleteWriter(
+      EncryptedOutputFile file, PartitionSpec spec, StructLike partition) {
+    EncryptionKeyMetadata keyMetadata = file.keyMetadata();
+    Map<String, String> properties = table.properties();
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+
+    try {
+      return DataFileServiceRegistry.writeBuilder(deleteFileFormat, inputType, file)
+          .set(properties)
+          .set(writeProperties)
+          .metricsConfig(metricsConfig)
+          .engineSchema(equalityDeleteRowSchemaType())
+          .withRowSchema(equalityDeleteRowSchema)
+          .withEqualityFieldIds(equalityFieldIds)
+          .withSpec(spec)
+          .withPartition(partition)
+          .withKeyMetadata(keyMetadata)
+          .withSortOrder(equalityDeleteSortOrder)
+          .overwrite()
+          .equalityDeleteWriter();
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to create new equality delete writer", e);
+    }
+  }
+
+  @Override
+  public PositionDeleteWriter<T> newPositionDeleteWriter(
+      EncryptedOutputFile file, PartitionSpec spec, StructLike partition) {
+    EncryptionKeyMetadata keyMetadata = file.keyMetadata();
+    Map<String, String> properties = table.properties();
+    MetricsConfig metricsConfig = MetricsConfig.forPositionDelete(table);
+
+    try {
+      return DataFileServiceRegistry.writeBuilder(deleteFileFormat, inputType, file)
+          .set(properties)
+          .set(writeProperties)
```

Comment:
This doesn't seem right. Why are table and write properties both being set?

Reply:
This is how it is done currently. Just the `set` calls are in different methods (`SparkFileWriterFactory` for the `writeProperties` and for the actual table properties

Reply:
It looks like this goes back a long way in history. I see it predates the new Spark writer classes, so we should probably leave it this way.